cg/apps/demultiplex/sample_sheet/index.py

```diff
@@ -156,19 +157,29 @@ def get_hamming_distance_index_1(sequence_1: str, sequence_2: str) -> int:
     )
 
 
-def get_hamming_distance_index_2(sequence_1: str, sequence_2: str) -> int:
+def get_hamming_distance_index_2(
+    sequence_1: str, sequence_2: str, is_reverse_complement: bool
+) -> int:
     """Get the hamming distance between two index 2 sequences.
     In the case that one sequence is longer than the other, the distance is calculated between
-    the shortest sequence and the last segment of equal length of the longest sequence."""
+    the shortest sequence and the last segment of equal length of the longest sequence.
+    If the sample requires reverse complement, the calculation is the same as for index 1."""
     shortest_index_length: int = min(len(sequence_1), len(sequence_2))
-    return get_hamming_distance(
-        str_1=sequence_1[-shortest_index_length:], str_2=sequence_2[-shortest_index_length:]
+    return (
+        get_hamming_distance(
+            str_1=sequence_1[:shortest_index_length], str_2=sequence_2[:shortest_index_length]
+        )
+        if is_reverse_complement
+        else get_hamming_distance(
+            str_1=sequence_1[-shortest_index_length:], str_2=sequence_2[-shortest_index_length:]
+        )
```

Comment:
Does this work? I interpret this as "start on the negative shortest_index_length and continue until the end", but maybe my Python knowledge is off? I was thinking it should be "length of sequence_1 - shortest_index_length" or something?

Reply:
This means exactly length of sequence_1 - shortest_index_length 😊 . The minus means you start counting from the tail to the head.